store/sqlstore/channel_store.go

```diff
@@ -334,6 +334,30 @@ type publicChannel struct {
 	Purpose     string `json:"purpose"`
 }
 
+type sidebarCategoryType string
+
+const (
+	SidebarCategoryChannels       sidebarCategoryType = "C"
+	SidebarCategoryDirectMessages sidebarCategoryType = "D"
+	SidebarCategoryFavorites      sidebarCategoryType = "F"
```

Comment:
I wanted to keep them as short as possible to avoid unneeded long strings in JSON responses... We do the same with channel types (O/P/F/G/D).
@hmhealey  Thoughts?

Reply:
I don't actually know the origin of the shorter names for channel types, so I don't know if that was the original reason for using them. My main concern with using the single-letter names is that it means someone looking at the API response directly may have to look up the meaning of the individual letters later on, especially if we run into a case like the channel types where Public and Private both start with the same letter. I also don't think that the shorter names would noticeably impact the size of API responses, especially if we gzip them (although I can't remember if that actually got implemented or not).

That said, I'm personally biased towards more verbose names, so I'm 0/5 either way. We just need to remember to update the client-side versions of these constants.